Add env descriptions to docs (#1059)
superfly/docs: reference/runtime-environment.html.md

### `FLY_IMAGE_REF`
**Docker Image Reference**: The name of the Docker image running this container. Useful if your application needs to launch Machine instances of itself to scale up-or-down. Read about how [a Rails app runs Machines for scale-to-zero background workers](https://fly.io/ruby-dispatch/rails-background-jobs-with-fly-machines/). `registry.fly.io/my-app-name:deployment-01H9RK9EYO9PGNBYAKGXSHV0PH` is an example of the Docker Image Reference's format.

### `FLY_MACHINE_ID`
**Machine ID**: The unique ID that flyctl commands and the Machines API use to identify a Machine. You'll also see it in the [logs](/docs/flyctl/logs/).

### `FLY_MACHINE_VERSION`

> **Machine Configuration Version**: Every Machine configuration has an associated version. Whenever you update a Machine's configuration (including when you update its Docker image), it gets a new `FLY_MACHINE_VERSION`. The exception to this is that changing only the Machine's metadata doesn't trigger a new version.

Reviewer: How about this? It's a bit lifeless, but the info is there.

```suggestion
**Machine Configuration Version**: Any update to a Machine's configuration, including its Docker image, generates a new `FLY_MACHINE_VERSION`. Updating metadata alone does not trigger a new version. `FLY_MACHINE_VERSION` has the same value as `Instance ID` in the output of `fly machine status`. 
```

The original wording echoed the structure of similar entries in the existing doc. This one needs a certain amount of explanation at this time, because it's not intuitive and as far as I'm aware it's not mentioned elsewhere. That can be corrected (and the doc as a whole could be tidied) at some time past the scope of this PR.

Author: No, this is great! I love how it's more descriptive. This was probably the hardest one for me to come up with information to talk/describe about.

> **Machine Configuration Version**: The version associated to a specific Machine configuration. When you update a Machine's configuration (including when you update its Docker image), it gets a new `FLY_MACHINE_VERSION`. You can see this change by running `fly machine status`. Changing the Machine's metadata however doesn't trigger a new version.

Reviewer: ```suggestion
**Machine Configuration Version**: The version associated to a specific Machine configuration. When you update a Machine's configuration (including when you update its Docker image), it gets a new `FLY_MACHINE_VERSION`. You can also find this value under the name `Instance ID` in the output of `fly machine status`. Changing the Machine's metadata however doesn't trigger a new version.
```

> **Machine Configuration Version**: Every Machine configuration has an associated version. Whenever you update a Machine's configuration (including when you update its Docker image), it gets a new `FLY_MACHINE_VERSION`. The exception to this is that changing only the Machine's metadata doesn't trigger a new version.

Reviewer 2: ```suggestion
**Machine Configuration Version**: The version associated to a specific Machine configuration. When you update a Machine's configuration (including when you update its Docker image), it gets a new `FLY_MACHINE_VERSION`. Changing the Machine's metadata doesn't trigger a new version.
```

Suggesting some simplification, but feel free to change. 

My big question is, where do users see this? Is is the "fly_release_version"? Or is the "Instance ID" that you see when you run `fly m status`? Instance ID seems to correspond to Machine config version in admin.

Author: Sorry I'm having a hard time understanding the question. Could you possibly rephrase it for me?

Reviewer: Good catch! @Lucais11 had in the draft that this is the Instance ID that you see in the output of `fly machine status`. It's my fault it got lost.

Author: Oh okay I get what was being asked now. Yes it is in the output of `fly machine status`.

### `FLY_PRIVATE_IP`

> **Private IPv6 Address**: This is the `IPv6 address` of the Machine on its [6PN private network](/docs/reference/private-networking/).

Reviewer 2: ```suggestion
**Private IPv6 Address**: The IPv6 address of the Machine on its [6PN private network](/docs/reference/private-networking/).
```

### `FLY_PROCESS_GROUP`

> **Process Group**: This is [the process group](/docs/apps/processes) associated with the Machine. 

Reviewer 2: ```suggestion
**Process Group**: The [process group](/docs/apps/processes) associated with the Machine. 
```

Does it make sense to say "the process group that the Machine belongs to"?

### `FLY_VM_MEMORY_MB`

> **Machine Memory**: This is the memory allocated to the Machine, in MBs. It's the same value you'll find under https://fly.io/dashboard/personal/machines and VM Memory in the output of `fly machine status`. More information on sizing [here](/docs/machines/guides-examples/machine-sizing/)!

Reviewer 2: ```suggestion
**Machine Memory**: The memory allocated to the Machine, in MB. It's the same value you'll find under https://fly.io/dashboard/personal/machines and VM Memory in the output of `fly machine status`. Learn more about [Machine sizing](/docs/machines/guides-examples/machine-sizing/).
```
